### PR TITLE
Make mode always u32 and switch to has_mode for StatRes fbs

### PR DESCRIPTION
**Pull request by kevinkassimo**
Using `int` as mode is a mistake (based on Go's [FileMode](https://golang.org/pkg/os/#FileMode)). Switch back to `uint` and uses `has_mode` to represent non-unix case

### CODE REVIEW

File: src/flags.rs [hunk 1]
```diff
@@ -222,7 +221,6 @@ pub fn v8_set_flags(args: Vec<String>) -> Vec<String> {
       let cstr = CStr::from_ptr(*ptr as *const i8);
       let slice = cstr.to_str().unwrap();
       slice.to_string()
-    })
-    .chain(rest.into_iter())
```
Comment (ry) [inline src/flags.rs]: Unrelated changes

Comment (kevinkassimo) [reply src/flags.rs]: Reverting them anyways for now
